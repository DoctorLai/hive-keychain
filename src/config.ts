const Config = {
  hiveEngine: {
    mainNet: 'ssc-mainnet-hive',
  },
  claims: {
    FREQUENCY: +(process.env.DEV_CLAIM_FREQUENCY || 10 * 60 * 1000),
    freeAccount: {
      MIN_RC_PCT: +(process.env.DEV_CLAIM_ACCOUNT_RC_PCT || 95),
      MIN_RC: +(process.env.DEV_CLAIM_ACCOUNT_MIN_RC || 1.1 * 10 * 10 ** 12), // 10% more than 10^13 (current creation cost)
    },
  },
};

export default Config;
