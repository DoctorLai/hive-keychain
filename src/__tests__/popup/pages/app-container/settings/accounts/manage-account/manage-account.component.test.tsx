import { screen } from '@testing-library/react';
import manageAccounts from 'src/__tests__/popup/pages/app-container/settings/accounts/manage-account/mocks/manage-accounts';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import alComponent from 'src/__tests__/utils-for-testing/aria-labels/al-component';
import alDiv from 'src/__tests__/utils-for-testing/aria-labels/al-div';
import alIcon from 'src/__tests__/utils-for-testing/aria-labels/al-icon';
import alSelect from 'src/__tests__/utils-for-testing/aria-labels/al-select';
import mk from 'src/__tests__/utils-for-testing/data/mk';
import { QueryDOM } from 'src/__tests__/utils-for-testing/enums/enums';
import assertion from 'src/__tests__/utils-for-testing/preset/assertion';
import config from 'src/__tests__/utils-for-testing/setups/config';
import {
  actAdvanceTime,
  clickAwait,
} from 'src/__tests__/utils-for-testing/setups/events';
config.byDefault();
describe('manage-account.component tests:\n', () => {
  let _asFragment: () => DocumentFragment;
  const { methods, constants, extraMocks, keys } = manageAccounts;
  const { snapshotName } = constants;
  methods.afterEach;
  describe('General cases:\n', () => {
    beforeEach(async () => {
      _asFragment = await manageAccounts.beforeEach();
    });
    it('Must display manage-account and match snapshot', () => {
      expect(_asFragment()).toMatchSnapshot(snapshotName.withData.default);
    });
    it('Must change to selected account', async () => {
      extraMocks.remockGetAccount();
      await clickAwait([
        alSelect.accountSelector,
        alSelect.itemSelectorPreFix + mk.user.two,
      ]);
      expect(screen.getByLabelText(alDiv.selectedAccount)).toHaveTextContent(
        mk.user.two,
      );
    });
    it('Must show selected private key when clicking', async () => {
      for (let i = 0; i < keys.length; i++) {
        await clickAwait([
          alDiv.keys.list.preFix.clickeableKey +
            methods.getKeyName(keys[i].keyName),
        ]);
        assertion.getOneByText(keys[i].privateKey);
      }
    });
    it('Must copy selected private key to clipboard', async () => {
      for (let i = 0; i < keys.length; i++) {
        await clickAwait([
          alDiv.keys.list.preFix.clickeableKey +
            methods.getKeyName(keys[i].keyName),
          alDiv.keys.list.preFix.clickeableKey +
            methods.getKeyName(keys[i].keyName),
        ]);
      }
      actAdvanceTime(200);
      expect(screen.queryAllByText(constants.message.copied)).toHaveLength(3);
    });
    it('Must show confirmation page and go back', async () => {
      for (let i = 0; i < keys.length; i++) {
        await clickAwait([
          alIcon.keys.list.preFix.remove + methods.getKeyName(keys[i].keyName),
        ]);
        assertion.getByLabelText(alComponent.confirmationPage);
        await clickAwait([alButton.dialog.cancel]);
      }
    });
    it.todo('Must show error if only one key');
    it.todo('Must show QR code');
    it.todo('Must remove from keychain');

    //maybe loading initial data? or just moving it to add-key component if gets too long
    it.todo('Must add all keys using master password');
    it.todo('Must add active key');
    it.todo('Must add memo key');
    it.todo('Must add posting key');

    //different data tests
    it.todo('Must not displayed remove button');
  });
  describe('Removing key cases:\n', () => {
    beforeEach(async () => {
      _asFragment = await manageAccounts.beforeEach();
    });
    it('Must remove active key', async () => {
      //TODO actual app take the user to the homepage after deleting a key.
      const ariaLabel =
        alIcon.keys.list.preFix.remove +
        methods.getKeyName('popup_html_active');
      await clickAwait([ariaLabel]);
      assertion.getByLabelText(alComponent.confirmationPage);
      await clickAwait([alButton.dialog.confirm]);
      await assertion.awaitFor(alComponent.homePage, QueryDOM.BYLABEL);
      await methods.gotoManageAccounts();
      assertion.queryByLabel(ariaLabel, false);
    });
    it('Must remove posting key', async () => {
      _asFragment = await manageAccounts.beforeEach();
      const ariaLabel =
        alIcon.keys.list.preFix.remove +
        methods.getKeyName('popup_html_posting');
      await clickAwait([ariaLabel]);
      assertion.getByLabelText(alComponent.confirmationPage);
      await clickAwait([alButton.dialog.confirm]);
      await assertion.awaitFor(alComponent.homePage, QueryDOM.BYLABEL);
      await methods.gotoManageAccounts();
      assertion.queryByLabel(ariaLabel, false);
    });
    it('Must remove memo key', async () => {
      _asFragment = await manageAccounts.beforeEach();
      const ariaLabel =
        alIcon.keys.list.preFix.remove + methods.getKeyName('popup_html_memo');
      await clickAwait([ariaLabel]);
      assertion.getByLabelText(alComponent.confirmationPage);
      await clickAwait([alButton.dialog.confirm]);
      await assertion.awaitFor(alComponent.homePage, QueryDOM.BYLABEL);
      await methods.gotoManageAccounts();
      assertion.queryByLabel(ariaLabel, false);
    });
  });
});
