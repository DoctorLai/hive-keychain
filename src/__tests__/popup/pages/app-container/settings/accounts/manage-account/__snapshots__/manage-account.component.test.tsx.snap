// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manage-account.component tests:
 General cases:
 Must display manage-account and match snapshot: manage-account.component ALL KEYS 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          Manage accounts
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          aria-label="settings-manage-account"
          class="settings-manage-account"
        >
          <div>
            <div
              class="select-account-section"
            >
              <div>
                <div
                  aria-expanded="false"
                  aria-label="Dropdown select"
                  class="react-dropdown-select select-account-select css-16vukcu e1gzf2xs0"
                  color="#0074D9"
                  direction="ltr"
                  tabindex="0"
                >
                  <div
                    class="react-dropdown-select-content react-dropdown-select-type-single css-1m5113o e1gn6jc30"
                  >
                    <div
                      class="selected-account-panel"
                    >
                      <img
                        src="https://images.hive.blog/u/keychain.tests/avatar"
                      />
                      <div
                        aria-label="selected-account-name"
                        class="selected-account-name"
                      >
                        keychain.tests
                      </div>
                    </div>
                  </div>
                  <div
                    class="react-dropdown-select-dropdown-handle css-ago8sv e1vudypg0"
                    color="#0074D9"
                    rotate="1"
                    tabindex="-1"
                  >
                    <svg
                      fill="currentColor"
                      viewBox="0 0 40 40"
                    >
                      <path
                        d="M31 26.4q0 .3-.2.5l-1.1 1.2q-.3.2-.6.2t-.5-.2l-8.7-8.8-8.8 8.8q-.2.2-.5.2t-.5-.2l-1.2-1.2q-.2-.2-.2-.5t.2-.5l10.4-10.4q.3-.2.6-.2t.5.2l10.4 10.4q.2.2.2.5z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="account-keys-list"
          >
            <div
              class="keys-panel"
            >
              <div
                class="account-keys-list-item"
              >
                <div
                  class="top-panel"
                >
                  <div
                    class="key-name"
                  >
                    Posting Key
                  </div>
                  <span
                    aria-label="icon-remove-key-Posting Key"
                    class="icon-component material-icons-outlined remove-button clickable"
                  >
                    delete
                  </span>
                </div>
                <div
                  class="keys-panel"
                >
                  <div
                    aria-label="clickeable-account-key-Posting Key"
                    class="private-key key-field hidden"
                  >
                    Click to show private key
                  </div>
                  <div
                    class="public-key key-field"
                  >
                    STM7KKUZb1CzwRiaN2RQcGeJUpcHM5BmCNudxXW21xqktBe91RpD8
                  </div>
                </div>
              </div>
              <div
                class="account-keys-list-item"
              >
                <div
                  class="top-panel"
                >
                  <div
                    class="key-name"
                  >
                    Active Key
                  </div>
                  <span
                    aria-label="icon-remove-key-Active Key"
                    class="icon-component material-icons-outlined remove-button clickable"
                  >
                    delete
                  </span>
                </div>
                <div
                  class="keys-panel"
                >
                  <div
                    aria-label="clickeable-account-key-Active Key"
                    class="private-key key-field hidden"
                  >
                    Click to show private key
                  </div>
                  <div
                    class="public-key key-field"
                  >
                    STM622dVYEu3Hiic6F4QiH1JALAXm8yRqfLd5kCq8xs8ZzNRnKGXC
                  </div>
                </div>
              </div>
              <div
                class="account-keys-list-item"
              >
                <div
                  class="top-panel"
                >
                  <div
                    class="key-name"
                  >
                    Memo Key
                  </div>
                  <span
                    aria-label="icon-remove-key-Memo Key"
                    class="icon-component material-icons-outlined remove-button clickable"
                  >
                    delete
                  </span>
                </div>
                <div
                  class="keys-panel"
                >
                  <div
                    aria-label="clickeable-account-key-Memo Key"
                    class="private-key key-field hidden"
                  >
                    Click to show private key
                  </div>
                  <div
                    class="public-key key-field"
                  >
                    STM8eALyQwyb2C4XhXJ7eZfjfjfSeNeeZREaxPcJRApie1uwzzcuF
                  </div>
                </div>
              </div>
            </div>
            <button
              aria-label="button-toogle-qr-code"
              class="submit-button default "
            >
              <div
                class="button-label"
              >
                Show QR Code 
              </div>
            </button>
            <button
              aria-label="button-delete-account"
              class="submit-button important "
            >
              <div
                class="button-label"
              >
                Remove from Keychain 
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;
